server/app/uk/gov/ons/addressIndex/server/modules/AddressIndexRepository.scala

```diff
@@ -695,6 +703,20 @@ class AddressIndexRepository @Inject()(conf: AddressIndexConfigModule,
       case None => Seq.empty
     }
 
+    val fromSourceQuery1 = args.fromSource match {
+      case "ewonly" => Seq(termsQuery("fromSource","EW"))
+      case "nionly" => Seq(termsQuery("fromSource","NI"))
+      case _ => Seq.empty
+    }
+
+    val fromSourceQuery2 = args.fromSource match {
+      case "ewonly" => Seq(termsQuery("fromSource","EW"))
+      case "nionly" => Seq(termsQuery("fromSource","NI"))
+      case "niboost" => Seq(termsQuery("fromSource","NI"))
+      case "ewboost" => Seq(termsQuery("fromSource","EW"))
+      case _ => Seq.empty
+    }
```

Comment:
These cases can be merged together where they overlap. `case "ewonly" | "ewboost" => ...`